example/ConquerClub.Domain/Army.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Text.RegularExpressions;
+
+namespace ConquerClub.Domain
+{
+    /// <summary>Represents an <see cref="Army"/></summary>
+    /// <remarks>
+    /// Equals is implemented including the owner (useful for unit tests).
+    /// The == and != use (like, <, >, <=, >=) <see cref="IComparable{Army}"/>
+    /// to allow the comparison of the size of the army only.
+    /// </remarks>
+    public readonly struct Army : IEquatable<Army>, IComparable<Army>, IComparable<int>, IFormattable
+    {
+        private static readonly Regex Pattern = new(@"^(?<Player>[A-Z0-9]+)\.Army\((?<Size>[0-9]+)\)$", RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        /// <summary>Gets no army.</summary>
+        public static readonly Army None;
+
+        /// <summary>Creates a new army.</summary>
+        internal Army(Player owner, int size)
+        {
+            Owner = owner;
+            Size = size;
```

Comment:
Can army sizes become negative and should this be allowed?

Reply:
It should not be allowed. Originally It had a guard (some of this code is borrowed from a private project of mine, creating a ConquerClub playing engine) for the public methods on the Army that changed the size. Now, if for some reason the result of an army is 0 (on a country) the validator fails, as all countries should have at least 1 army.

But given that we just argued that the Army is an SVO that should not allow an invalid state of its own, I agree that I should put back the Guard here.